exporter/azuremonitorexporter/metric_to_envelopes.go

```diff
@@ -0,0 +1,232 @@
+// Copyright OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azuremonitorexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/azuremonitorexporter"
+
+import (
+	"time"
+
+	"github.com/microsoft/ApplicationInsights-Go/appinsights/contracts"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+)
+
+type metricPacker struct {
+	logger *zap.Logger
+}
+
+type timedMetricDataPoint struct {
+	dataPoint *contracts.DataPoint
+	timestamp pcommon.Timestamp
+}
+
+type metricTimedData interface {
+	getTimedDataPoints() []*timedMetricDataPoint
+}
+
+func (packer *metricPacker) MetricToEnvelopes(metric pmetric.Metric) []*contracts.Envelope {
+	var envelopes []*contracts.Envelope
+
+	mtd := packer.getMetricTimedData(metric)
+
+	if mtd != nil {
+
+		for _, timedDataPoint := range mtd.getTimedDataPoints() {
+
+			envelope := contracts.NewEnvelope()
+			envelope.Tags = make(map[string]string)
+			envelope.Time = toTime(timedDataPoint.timestamp).Format(time.RFC3339Nano)
+
+			metricData := contracts.NewMetricData()
+			dataPoint := timedDataPoint.dataPoint
+			metricData.Metrics = []*contracts.DataPoint{dataPoint}
+			metricData.Properties = make(map[string]string)
```

Comment:
You might need to add the standard tags for the cloud role/cloud role instance, instrumentation library properties, etc. on this. Maybe something similar to here:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/9bc3a8ba08ca2f58a12c16a9d4a5ee0f80ce21d4/exporter/azuremonitorexporter/trace_to_envelope.go#L123